datacube_alchemist/worker.py

```diff
@@ -125,10 +125,17 @@ def _transform_with_args(self, task: AlchemistTask) -> Type[Transformation]:
     def _find_dataset(self, uuid: str) -> Dataset:
         # Find a dataset for a given UUID from within the available
         dataset = self.dc.index.datasets.get(uuid)
-        if dataset.type in self.input_products:
-            return dataset
+        if dataset is not None:
+            if dataset.type not in self.input_products:
+                # Dataset is in the wrong product
+                dataset = None
```

Comment:
Not sure about returning None and logging an error. Maybe it should throw an exception or do something noisier.

Reply:
Yeah, I'm not sure either. Datacube returns None, though. And it is handled in the CLI appropriately. I'll see how it plays out.